Load full PR if it is missing required fields

bluekeyes opened the pull request:
The PR objects included in webhook payloads are oddly inconsistent: the
pull_request_review payload does not include the "commits" field, which
is require for policy-bot to function. Instead of loading PRs in each
handler, modify the context to load the PR if the one it was given is
missing any required fields. To make this harder to mess up, use a new
pull request object internally based on the GraphQL query.

Fixes #63.

bluekeyes committed:
Load full PR if it is missing required fields

The PR objects included in webhook payloads are oddly inconsistent: the
pull_request_review payload does not include the "commits" field, which
is require for policy-bot to function. Instead of loading PRs in each
handler, modify the context to load the PR if the one it was given is
missing any required fields. To make this harder to mess up, use a new
pull request object internally based on the GraphQL query.

bluekeyes commented:
The next step here would be to refactor `NewGitHubContext` to take something that minimally provides the locator information but might be cast to a full `github.PullRequest`. That would save us from loading the pull request explicitly in the `issue_comment` handler.

ajlake commented on pull/github.go:
```diff
@@ -103,7 +135,11 @@ func (ghc *GitHubContext) RepositoryName() string {
 func (ghc *GitHubContext) Author() (string, error) {
-	return ghc.pr.GetUser().GetLogin(), nil
+	pr, err := ghc.get()
+	if err != nil {
```
Should we just load the PR at construction time? Maybe `v3PullRequestToV4` can also take in the Github client necessary to populate the `v4` object if there is insufficient information in the v3 object, instead of returning `nil`.

In that case, we'd probably want to move the `case` checks into a helper:
```
v3PullRequestToV4(...):
    if isSufficient(v3) {
      // seed from v4
    }

    // else graphql query
```

bluekeyes replied:
I introduced the concept of a `Locator` which contains an optional `github.PullRequest`. If that is missing or incomplete, the full value is now loaded at construction time. I also split out the completeness test as you suggested.

bluekeyes committed:
Merge branch 'develop' into bkeyes/load-pr

bluekeyes committed:
Load pull request at context creation

Introduce a Locator type to hold the identifying information for a PR
and an optional PR object. If the PR is missing or is not complete, make
an API request at creation time for the missing data. Now that this data
is guaranteed, remove the error return values from methods that only use
PR data.

Also refactor how the base handler evaluates PRs to reduce the number of
arguments and to pass pull.Context values instead of PR objects.
